packages/long-short-pair/src/mappings/longShortPair.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  getOrCreateLongShortPairContract,
+  getOrCreateToken,
+  getOrCreateTokensCreatedEvent,
+  getOrCreateTokensRedeemedEvent,
+  getOrCreateContractExpiredEvent,
+  getOrCreatePositionSettledEvent,
+} from "../utils/helpers";
+
+import {
+  LongShortPair,
+  TokensCreated,
+  TokensRedeemed,
+  ContractExpired,
+  PositionSettled,
+} from "../../generated/templates/LongShortPair/LongShortPair";
+
+import { toDecimal } from "../utils/decimals";
+import { log, Address, BigInt } from "@graphprotocol/graph-ts";
+
+function updateLSP(
+  lspAddress: Address,
+  collateralChange: BigInt,
+  longTokensChange: BigInt,
+  shortTokensChange: BigInt,
+  tokenCollateralAddition: boolean = true,
```

Comment:
Thoughts on setting the sign of `collateralChange`, `longTokensChange`, and `shortTokensChange` instead of using the `tokenCollateralAddition` flag. This could allow you to remove the `else` statement from below. And then you can use BigInt/BigDecimal syntax from [here](https://thegraph.com/docs/assemblyscript-api#built-in-types)

Reply:
I agree that'll be better. I tried this but had issues with it. I was trying to use something like this
```
collateralChange.times(new BigInt(-1))
etc...
```

but this failed with an unknown error. I'll try toggle the polarity again with different syntax and see if I can get it to work.